submissions/f3d0t/friends-app/app.js

```diff
@@ -0,0 +1,173 @@
+const RESET_BUTTON = document.querySelector(".filters__button");
+const DAY_NIGHT = document.querySelector(".day_night");
+const FRIENDS_COUNT = 60;
+const REQUEST_LINK = `https://randomuser.me/api/?results=${FRIENDS_COUNT}`;
+
+const fetchData = async (requestLink) => {
+	try {
+		const response = await fetch(requestLink);
+		if (!response.ok) {
+			throw Error("HTTP-Error: " + response.statusText);
+		}
+		const { results } = await response.json();
+		return results;
+	} catch (error) {
+		alert(error);
+	}
+};
+
+const initApp = (friendsDataArray) => {
+	const friends = new FriendsList(friendsDataArray, MAIN);
+	friends.renderCards();
+	bindEventListeners(friends);
+};
+
+const bindEventListeners = (friends) => {
+	FILTERS.addEventListener("input", ({ target }) => {
+		if (target.name === "search") {
+			friends.filterBySearch(target.value);
+		}
+		if (target.name === "gender") {
+			friends.filterByGender(target.id);
+			FILTERS.elements.search.value = "";
+			const checkedUserName = FILTERS.querySelector("[name=userName]:checked");
+			if (checkedUserName) {
+				friends.sortByName(checkedUserName.id);
+			}
+			const checkedUserAge = FILTERS.querySelector("[name=userAge]:checked");
+			if (checkedUserAge) {
+				friends.sortByAge(checkedUserAge.id);
+			} //this two forEach callbacks is needed to re-sort new arrays of cards, filtered by gender
+		}
+		if (target.name === "userName") {
+			friends.sortByName(target.id);
+			FILTERS.elements.search.value = "";
+			const checkedUserAge = FILTERS.querySelector("[name=userAge]:checked");
+			if (checkedUserAge) {
+				checkedUserAge.checked = false;
+			}
+		}
+		if (target.name === "userAge") {
+			friends.sortByAge(target.id);
+			FILTERS.elements.search.value = "";
+		}
+	});
+	RESET_BUTTON.addEventListener("click", (e) => {
+		e.preventDefault();
+		FILTERS.querySelectorAll("input[type=radio]:checked").forEach((radioButton) => {
+			radioButton.checked = false;
+		});
+		FILTERS.elements.search.value = "";
+		friends.resetFilters();
+	});
+	FILTERS.addEventListener("submit", (event) => {
+		if (event.key === "Enter") {
+			event.preventDefault();
+		}
+	});
+};
+
+class FriendsList {
+	constructor(data, parentWrapper) {
+		this.allCards = data.map((friendData) => new FriendCard(friendData));
+		this.currentCards = [...this.allCards];
+		this.wrapper = parentWrapper;
+	}
+	renderCards(cardsArray = this.currentCards) {
+		this.wrapper.innerHTML = "";
+		this.wrapper.append(...cardsArray.map((card) => card.createCard()));
+	}
+	resetFilters() {
+		this.currentCards = [...this.allCards];
+		this.renderCards();
+	}
+	filterByGender(gender) {
+		if (gender === "male") {
+			this.currentCards = this.allCards.filter((friendCard) => friendCard.gender === "male");
+		}
+		if (gender === "female") {
+			this.currentCards = this.allCards.filter((friendCard) => friendCard.gender === "female");
+		}
+		if (gender === "all") {
+			this.currentCards = [...this.allCards];
+		}
```

Comment:
```suggestion
		if (gender === "all") {
			this.currentCards = [...this.allCards];
		} else {
		        this.currentCards = this.allCards.filter((friendCard) => friendCard.gender === gender );
		}
```

Reply:
How big value of tab spaces you have? Looks like more than 4. If so, please use one of the two most commonly used - 2 or 4.

Reply:
I have 2 tab space width in my VS code

Reply:
Do you mean that you use tabs, not spaces? Because indentation larger than 2 spaces.